google/cloud/aiplatform/training_jobs.py

```diff
@@ -1352,7 +1358,15 @@ def _prepare_and_validate_run(
             accelerator_type=accelerator_type,
             boot_disk_type=boot_disk_type,
             boot_disk_size_gb=boot_disk_size_gb,
-        ).pool_specs
+        )
+
+        if reduction_server_replica_count > 0:
+            worker_pool_specs = worker_pool_specs.reduction_server_pool(
+                reduction_server_replica_count=reduction_server_replica_count,
+                reduction_server_machine_type=reduction_server_machine_type,
+            )
+
+        worker_pool_specs = worker_pool_specs.pool_specs
```

Comment:
This changes the type of the variable. Although allowable in Python, prefer not to do that for readability purposes.